shopinvader_api_cart/routers/cart.py

```diff
@@ -0,0 +1,210 @@
+        if float_compare(delta_qty, 0, precision_rounding=product_uom.rounding) <= 0:
+            return None
+        partner = cart.partner_id
+        vals = {
+            # Order in this dict is important and must be kept.
+            # When creating the transaction we call play_onchanges().
+            # These onchanges will be played following the order defined here.
+            # All computes depending on order_id must be triggered first,
+            # to set the currency on the SOL for e.g.
+            "order_id": cart.id,
+            "product_id": product_id,
+            "product_uom_qty": delta_qty,
+        }
+        ctx_lang = self.env.context.get("lang", partner.lang)
+        if partner.lang != ctx_lang:
+            product_id = vals["product_id"]
+            vals["name"] = self._get_sale_order_line_name(product_id)
+        return vals
+
+    @api.model
+    def _apply_transactions(self, cart, transactions: list[CartTransaction]):
+        """Apply transactions to the given cart."""
+        if not transactions:
+            return
+        cart.ensure_one()
+        self._check_transactions(transactions=transactions)
+        transactions_by_product_id = self._group_transactions_by_product_id(
+            transactions=transactions
+        )
+        update_cmds = []
+        with self.env.norecompute():
+            # prefetch all products
+            self.env["product.product"].browse(transactions_by_product_id.keys())
+            # here we avoid that each on change on a line trigger all the
+            # recompute methods on the SO. These methods will be triggered
+            # by the orm into the 'write' process
+            for product_id, trxs in transactions_by_product_id.items():
+                line = cart._get_cart_line(product_id)
+                if line:
+                    cmd = self._apply_transactions_on_existing_cart_line(line, trxs)
+                else:
+                    cmd = self._apply_transactions_creating_new_cart_line(cart, trxs)
+                if cmd:
+                    update_cmds.append(cmd)
+        all_transaction_uuids = transaction_uuids = [
+            t.uuid for t in transactions if t.uuid
+        ]
+        if cart.applied_cart_api_transaction_uuids:
+            all_transaction_uuids = [
+                cart.applied_cart_api_transaction_uuids
+            ] + transaction_uuids
+        vals = {"applied_cart_api_transaction_uuids": ",".join(all_transaction_uuids)}
+        if update_cmds:
+            vals["order_line"] = update_cmds
+        cart.write(vals)
+
+    @api.model
+    def _sync_cart(
+        self,
+        partner: ResPartner,
+        cart: SaleOrder,
+        uuid: str,
+        transactions: list[CartTransaction],
+    ):
+        if not cart and transactions:
+            cart = self.env["sale.order"]._create_empty_cart(partner.id)
+        if not uuid or cart.uuid == uuid:
```

Comment:
we can reach this line with no cart so cart.uuid may not do what it should?

Reply:
If no transaction, I'll directly return cart, as first step of the method. So if no cart, one new cart will always be created.